src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -2021,7 +2028,16 @@ private static function createRequestFromFactory(array $query = array(), array $
      */
     public function isFromTrustedProxy()
     {
-        return self::$trustedProxies && IpUtils::checkIp($this->server->get('REMOTE_ADDR'), self::$trustedProxies);
+        if (!self::$trustedProxies) {
+            return false;
+        }
+
+        $remoteAddr = $this->server->get('REMOTE_ADDR');
+        if (!array_key_exists($remoteAddr, self::$validTrustedProxyIP) || null === self::$validTrustedProxyIP[$remoteAddr]) {
```

Comment:
Can be simply: `!isset(self::$validTrustedProxyIP[$remoteAddr])` cause you do check for `null` anyway.